src/main/_docs/use-che-as-an-ide/ide-debug.md

```diff
@@ -187,3 +191,55 @@ Among other features, it also has the capability to trigger a debug session:
 ![z-ray-debug.png]({{ base }}{{ site.links["z-ray-debug.png"] }})
 
 That's all!
+
+# NodeJS
+
+The Node.js ready-to-go [stack]({{ base }}{{ site.links["ws-stacks"] }}) comes with a Node.js debugger module installed and configured. The Dockerfile is located in the [eclipse/che-dockerfiles](https://github.com/eclipse/che-dockerfiles/blob/master/recipes/node/latest/Dockerfile) repository.
+
+The debugging workflow is:
+
+1. Launch the Node.js debugger client to start a debug session
+1. Create/Run command to generate a preview URL
+1. Click the preview URL to interact with the app
+1. Use the debugger panel to perform debug functions
+
+You can set breakpoints in the editor at any time by clicking on the line number.
+
+## Starting the Node.js Debugger Client
+
+{{site.product_formal_name}} has the Node.js client integrated in the web IDE. to launch the debugger client:
+
+1. Go to `Run > Edit Debug Configurations` from the main menu
+1. Create a new `NODEJS` configuration
+1. Change any settings if necessary. The defaults are usually OK
+1. Click the `Debug` button
+1. The debugger will break at first line of code
+
+![debug-nodejs-config.png]({{ base }}{{ site.links["debug-nodejs-config.png"] }})
+
+### Creating a Command to View the Preview URL
+
+{{site.product_formal_name}}'s workspaces have machine(s) that are docker container(s). These workspace machines have exposed ports which are initially set by [stack]({{base}}{{site.links["ws-stacks"]}}) and [additional exposed ports](#exposed-ports) can be added through the [machine information in dashboard]({{base}}{{site.links["ws-machines"]}}#dashboard-machine-information)
```

Comment:
This mostly duplicates the ## Exposed Ports section at the bottom maybe we should just refer to that section instead?